wicker/plugins/wandb.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Any, Dict, Literal
+
+import wandb
+
+from wicker.core.config import get_config
+from wicker.core.definitions import DatasetID
+from wicker.core.storage import S3PathFactory
+
+
+def version_dataset(
+    dataset_name: str,
+    dataset_version: str,
+    metadata: Dict[str, Any],
+    dataset_backend: Literal["s3"] = "s3",
+) -> None:
+    """
+    Version the dataset on Weights and Biases using the config parameters defined in wickerconfig.json.
+
+    Args:
+        dataset_name: The name of the dataset to be versioned
+        dataset_version: The version of the dataset to be versioned
+        metadata: The metadata to be logged as an artifact, enforces dataclass for metadata documentation
+        dataset_backend: The backend where the dataset is stored, currently only supports s3
+    """
+    # needs to first acquire and set wandb creds
+    # WANDB_API_KEY, WANDB_BASE_URL, WANDB_USER_EMAIL
+    _acquire_wandb_credentials()
+
+    # needs to init the wandb run, this is going to be a 'data' run
+    dataset_run = wandb.init(project=f"{dataset_name}_curation", name=dataset_name)
```

Comment:
hmm, WDYT about having one project for all datasets, but perhaps with different properties? Also could `dataset_name` be duplicated for different curations?

Reply:
I like that idea, I'll change the project to `dataset_curation` if that sounds good to you.
It could be duplicated, the documentation is fuzzy on it but the only thing that changes in the portal is the artifacts that are uploaded. I thought it would make sense for the runs to all be named the global dataset with the artifacts having the version name. Thinking about it more I'll change it so that the version is also included in this.